crypto/params.c

```diff
@@ -895,3 +897,152 @@ OSSL_PARAM OSSL_PARAM_construct_end(void)
 
     return end;
 }
+
+static void param_process(va_list ap, const char *name,
+                          void (*f)(const char *name, int type, size_t len,
+                                    const void *value, void *arg, int bignum),
+                          void *arg)
+{
+    int type;
+
+    for (; name != NULL; name = va_arg(ap, const char *)) {
+        type = va_arg(ap, int);
+#define T(type, type_name, param_type) \
```

Comment:
maybe make this be ```CASE``` instead of ```T```

Reply:
Looks kind of ugly, too many leading capitals after argument reordering.

Reply:
yeah, but then it looks kinda like C code.